packages/runtime-core/src/componentEmits.ts

```diff
@@ -269,8 +269,11 @@ export function isEmitListener(
   if (__COMPAT__ && key.startsWith(compatModelEventPrefix)) {
     return true
   }
-
-  key = key.slice(2).replace(/Once$/, '')
+  key = key.slice(2)
+  // #8342
+  // e.g. emits: { once: null } -> onOnce、ononce
+  // Does not require regular expression processing
+  key = key.toLowerCase() === 'once' ? key : key.replace(/Once$/, '')
```

Comment:
Maybe this?

```suggestion
  key = key.replace(/Once$/, '') || key
```

While it is less explicit, it avoids the `toLowerCase()` call.